.circleci/config.yml

```diff
@@ -11,11 +15,62 @@ executors:
     docker:
     - image: circleci/golang:1.13
 
+commands:
+  publish_image:
+    parameters:
+      registry:
+        description: Registry address
+        type: string
+        default: "docker.io"
+      organization:
+        description: Registry organization
+        type: string
+        default: prombench
+      image_name:
+        description: Name of the container image
+        type: string
+      login:
+        description: Environment variable holding the registry login
+        type: env_var_name
+        default: DOCKER_USER
+      password:
+        description: Environment variable holding the registry password
+        type: env_var_name
+        default: DOCKER_PASSWORD
+      dockerfile_path:
+        description: Path to Dockerfile
+        type: string
+        default: ""
+      context_path:
+        description: Path of docker build context
+        type: string
+        default: "."
+    steps:
+    - run:
+        name: Setup environment variables
+        command: |
+          echo "export TAG=`date +%Y%m%d`" >> $BASH_ENV
+          echo "DOCKER_REPO=<< parameters.registry >>/${DOCKER_ORG:-<< parameters.organization >>}" >> $BASH_ENV
+          echo "DOCKER_IMAGE=<< parameters.image_name >>" >> $BASH_ENV
+    - when:
+        condition: <<parameters.dockerfile_path>>
+        steps:
+        - run: docker build -t $DOCKER_REPO/$DOCKER_IMAGE:$TAG -f << parameters.dockerfile_path >> << parameters.context_path >>
+    - unless:
+        condition: <<parameters.dockerfile_path>>
+        steps:
+        - run: docker build -t $DOCKER_REPO/$DOCKER_IMAGE:$TAG << parameters.context_path >>
+    - run:
+        name: Push images to registry
+        command: |
+          docker images
```

Comment:
do we need this?

Reply:
This helps for debugging.
To make sure the image is created as we expected, i.e. with the CORRECT image name, tag.

Reply:
ok